include/xtensor/xscalar.hpp

```diff
@@ -455,12 +459,30 @@ namespace xt
      * xscalar implementation *
      **************************/
 
+    template <class CT>
+    inline xscalar<CT>::xscalar() noexcept
```

Comment:
This should be disabled when CT is a reference type.

Reply:
Note for the record: we found that the error when you try default-instantiating `xscalar` with a reference type was nicer without the `enable_if` than with it.

Reply:
I added a little comment to the implementation